### ISSUE TITLE
redis-7.0.4 build failure

### ISSUE THREAD

**Issue by yayayayaka** [NixOS/nixpkgs#182088]
### Steps To Reproduce
1. build redis on master

### Build log
[build.log](https://github.com/NixOS/nixpkgs/files/9140853/build.log)

### Additional context
Discovered by building the nextcloud NixOS test in #182067

### Notify maintainers

@berdario @globin @marsam

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.18.10, NixOS, 22.05 (Quokka), 22.05pre-git`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.8.1`
 - nixpkgs: `/etc/src/nixpkgs`
```

**Comment by ajs124** [comment]
I think you've run into some flaky tests. I saw the same, but the builds seem to succeed on hydra.

Which is something still worth looking in to, but probably not an immediate problem.

**Comment by abathur** [comment]
I took a quick look at whether this had been fixed in the interim. I suspect probably not, but I did notice an upstream commit (https://github.com/redis/redis/commit/997fa41e99271cc5c3a79e9bf8a1332b3d9ab0c2) about test flakiness that happens to touch the very test mentioned in the attached log (`*** [err]: replica buffer don't induce eviction in tests/unit/maxmemory.tcl`). 

It's included in the `7.2-rc2` tag, so if this failure was vaguely reproducible it might be worth giving it a shake.
